File: backend/src/main/java/com/example/learninganalysis/service/FeedbackService.java
package com.example.learninganalysis.service;

import com.example.learninganalysis.dto.FeedbackResponse;
import com.example.learninganalysis.entity.Feedback;

import java.util.List;

public interface FeedbackService {
    
    /**
     * 提交学习反馈
     * 
     * @param studentId 学生ID
     * @param feedback 反馈内容
     * @return 是否提交成功
     */
    boolean submitFeedback(Long studentId, String feedback);
    
    /**
     * 获取学生的反馈列表
     * 
     * @param studentId 学生ID
     * @return 反馈列表
     */
    List<Feedback> getFeedbackByStudentId(Long studentId);
    
    /**
     * 教师回复学生反馈
     * 
     * @param feedbackId 反馈ID
     * @param reply 回复内容
     * @return 是否回复成功
     */
    boolean replyToFeedback(Long feedbackId, String reply);
    
    /**
     * 获取学生的反馈（教师视图）
     */
    List<FeedbackResponse> getStudentFeedbacks(Long studentId, String token);
    
    /**
     * 获取我的反馈
     */
    List<FeedbackResponse> getMyFeedbacks(String token);
    
    /**
     * 回复反馈
     */
    void respondToFeedback(Long feedbackId, String response, String token);
} 

File: frontend/src/services/feedbackService.ts
import axios from 'axios';

const API_URL = '/api';

// 提交反馈
export const submitFeedback = async (token: string, feedback: string) => {
  const response = await axios.post(`${API_URL}/feedback/submit`, {
    token,
    feedback
  });
  return response.data;
};

// 获取我的反馈列表
export const getMyFeedbacks = async (token: string) => {
  const response = await axios.get(`${API_URL}/feedback/list`, {
    params: { token }
  });
  return response.data;
};

// 获取学生反馈列表（教师用）
export const getStudentFeedbacks = async (token: string, studentId: number) => {
  const response = await axios.get(`${API_URL}/teacher/student/${studentId}/feedbacks`, {
    params: { token }
  });
  return response.data;
};

// 回复反馈
export const respondToFeedback = async (token: string, feedbackId: number, response: string) => {
  const res = await axios.post(`${API_URL}/feedback/${feedbackId}/respond`, {
    token,
    response
  });
  return res.data;
}; 